Don't throw "restartmanager canceled" error for no restart policy container

coolljt0725 opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/docker/docker/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

For additional information on our contributing process, read our contributing
guide https://docs.docker.com/opensource/code/

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**\- What I did**
Don't throw "restartmanager canceled" error for no restart policy container
or else if kill a no restart policy container, there is a error log 
`ERRO[0024] restartmanager canceled`

and add the container id to the warning message if a container has restart policy
and make error log to warn log

ping @tonistiigi 
**\- How I did it**

**\- How to verify it**

**\- A picture of a cute animal (not mandatory but encouraged)**

Signed-off-by: Lei Jitang leijitang@huawei.com

cpuguy83 commented on libcontainerd/container_linux.go:
```diff
@@ -120,7 +121,11 @@ func (ctr *container) handleEvent(e *containerd.Event) error {
 			restart, wait, err := ctr.restartManager.ShouldRestart(e.Status, false)
 			if err != nil {
-				logrus.Error(err)
+				if strings.Contains(err.Error(), "restartmanager canceled") {
```
Can we instead always wrap the error message?

coolljt0725 commented:
@cpuguy83  updated

cpuguy83 commented:
`libcontainerd/container_linux.go:8: imported and not used: "strings"`

coolljt0725 committed:
Don't throw "restartmanager canceled" error for no restart policy container

Don't throw "restartmanager canceled" error for no restart policy container
and add the container id to the warning message if a container has restart policy
and has been canceled.

Signed-off-by: Lei Jitang <leijitang@huawei.com>

thaJeztah commented:
ping @cpuguy83 @tonistiigi PTAL

crosbymichael commented:
LGTM

cpuguy83 commented:
LGTM
